paperqa/llms.py

```diff
@@ -65,9 +65,26 @@ class LiteLLMEmbeddingModel(EmbeddingModel):
     name: str = Field(default="text-embedding-3-small")
     embedding_kwargs: dict = Field(default_factory=dict)
 
+    def _truncate_if_large(self, texts: list[str]) -> list[str]:
+        """Truncate texts if they are too large specifically for an openai model."""
+        if self.name not in {
+            "text-embedding-3-small",
+            "text-embedding-3-large",
+            "text-embedding-ada-002",
+        }:
+            return texts
+        max_tokens = 8192
```

Comment:
we should probably move this and the model name list to module level variables

Reply:
Fine - I wrote it correctly using litellm cost map